components/compose/cfr/src/main/java/mozilla/components/compose/cfr/CFRPopupFullscreenLayout.kt

```diff
@@ -0,0 +1,381 @@
+    private val windowManager = anchor.context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+
+    /**
+     * Listener for when the anchor is removed from the screen.
+     * Useful in the following situations:
+     *   - lack of purpose - if there is no anchor the context/action to which this popup refers to disappeared
+     *   - leak from WindowManager - if removing the app from task manager while the popup is shown.
+     *
+     * Will not inform client about this since the user did not expressly dismissed this popup.
+     */
+    private val anchorDetachedListener = ViewDetachedListener {
+        dismiss()
+    }
+
+    /**
+     * When the screen is rotated the popup may get improperly anchored
+     * because of the async nature of insets and screen rotation.
+     * To avoid any improper anchorage the popups are automatically dismissed.
+     *
+     * Will not inform client about this since the user did not expressly dismissed this popup.
+     */
+    private val orientationChangeListener = DisplayOrientationListener(anchor.context) {
+        dismiss()
+    }
+
+    override var shouldCreateCompositionOnAttachedToWindow: Boolean = false
+        private set
+
+    init {
+        ViewTreeLifecycleOwner.set(this, ViewTreeLifecycleOwner.get(anchor))
+        this.setViewTreeSavedStateRegistryOwner(anchor.findViewTreeSavedStateRegistryOwner())
+        anchor.addOnAttachStateChangeListener(anchorDetachedListener)
+        orientationChangeListener.start()
+    }
+
+    /**
+     * Add a new CFR popup to the current window overlaying everything already displayed.
+     * This popup will be dismissed when the user clicks on the "x" button or based on other user actions
+     * with such behavior set in [CFRPopupProperties].
+     */
+    fun show() {
+        windowManager.addView(this, createLayoutParams())
+    }
+
+    @Composable
+    override fun Content() {
+        val anchorLocation = IntArray(2).apply {
+            anchor.getLocationOnScreen(this)
+        }
+
+        val anchorXCoordMiddle = Pixels(anchorLocation.first() + anchor.width / 2)
+        val indicatorArrowHeight = Pixels(
+            CFRPopup.DEFAULT_INDICATOR_HEIGHT.dp.toPx(),
+        )
+
+        val popupBounds = computePopupHorizontalBounds(
+            anchorMiddleXCoord = anchorXCoordMiddle,
+            arrowIndicatorWidth = Pixels(Companion.getIndicatorBaseWidthForHeight(indicatorArrowHeight.value)),
+        )
+
+        println("Mugurel: anchor start?: ${anchor.x}")
+        println("Mugurel: popup start?: ${popupBounds.startCoord}")
+
+        val indicatorOffset = computeIndicatorArrowStartCoord(
+            anchorMiddleXCoord = anchorXCoordMiddle,
+            popupStartCoord = popupBounds.startCoord,
+            arrowIndicatorWidth = Pixels(
+                Companion.getIndicatorBaseWidthForHeight(indicatorArrowHeight.value),
+            ),
+        )
+
+        MaterialTheme {
```

Comment:
I'm not well versed in Compose, so disregard accordingly: how does the `MaterialTheme` interact with our `FirefoxTheme` if we have it wrapped here?